### PR TITLE
[Wave 8] [Ideal Nav] goBack update fallback route

### PR DESCRIPTION
**Pull request by adamgrzybowski**
### Details
This solution will make navigation more stable before a proper rewrite ([issue](https://github.com/Expensify/App/issues/35938)) of the goBack function. This function is not easy to use and debug which may lead to more bugs in the future.

When I was going through all the reported ([issues](https://github.com/Expensify/App/issues/35626)) I noticed that they are fixed by removing ROUTES.HOME from the argument.

It looks like the new nested HOME route is not working well with the old goBack function with HOME as an argument. 

To prevent more bugs that are not yet found I decided to replace all cases where goBack is used with HOME argument.

It should prevent all critical problems like messed up navigation that needs to be reloaded. Now the worst case scenario if any is the wrong page after going back or simply going back a bit too much. 

**NOTE** Please have in mind that this PR is related to the goBack function. That means actions triggered from the page e.g arrow left rendered on the page or submit button. The go back arrow on the browser navigation bar or physical device is a separate issue.

**NOTE 2** There is one issue that seems to have different root cause https://github.com/Expensify/App/issues/35689. I will take care of that separately @hayata-suenaga 
### Fixed Issues

$ https://github.com/Expensify/App/issues/35626
PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

**Test 1**
1. Make sure you're on a mobile phone or on a browser with a smaller width
2. Click the Chat button on the bottom tab
3. Select a chat/report
4. Go back
5. Confirm that the Chat List page opens from the right when the page should close to the left (watch the video)
6. Click the Settings (🔧) button on the bottom tab
7. Confirm that the Setting page opens from the right when the page should not animate. Only the part above the bottom tab should change without animation.

**Test 2**
1. Click the Settings button on the bottom tab > Workspaces
2. Click the back button (left chevron icon)
3. Confirm that the Chat List page opens from the right when the Workspace List page should close to left to go back to the Settings page

**Test 3**
1. Test this on a browser. Shrink the window width.
2. Visit https://dev.new.expensify.com:8082/settings/preferences
3. Press the back button on the header (left chevron icon next to Preferences)
4. The app navigates to the Chats page when it should navigate back to the Settings page.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/67908363/ecc07f19-fc01-478d-a0d2-f78f142b4903

https://github.com/Expensify/App/assets/67908363/19222a39-84aa-4412-9fd4-51ab499a96b9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

I have a problem with android web setup. I will update video tomorrow.

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/67908363/c753a1d4-c759-4055-b073-bb269af58776

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/67908363/d8201ec6-8bb6-4436-b414-4262da34b227

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/67908363/1f6516ad-f9c2-4bba-8825-b80f89501f71

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/67908363/f466a85f-d2d6-4fd1-8e88-beb8a82ceb3e

</details>

### CODE REVIEW

**Comment by hayata-suenaga** [comment]
cc: @mountiny if you're also interested reviewing the PR 🙇

**Comment by getusha** [comment]
☑️ https://github.com/Expensify/App/issues/35669

https://github.com/Expensify/App/assets/75031127/7f458620-c76a-4d64-b65c-ad1aeaf85c16

**Comment by getusha** [comment]
@adamgrzybowski this issue is still reproducible https://github.com/Expensify/App/issues/35756

https://github.com/Expensify/App/assets/75031127/e4dd589d-ea51-4bad-a528-09d43f8a4c9a

**Comment by adamgrzybowski** [comment]
Hey @getusha  @hayata-suenaga, I made a comment here https://github.com/Expensify/App/issues/35756#issuecomment-1932214308. I think this should be a separate issue. It's not really related to goBack function.

**Comment by getusha** [comment]
Ah i missed that, i agree @hayata-suenaga i think we can reopen the issue.

**Comment by getusha** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by getusha** [comment]
https://github.com/Expensify/App/assets/75031127/44eae2fb-9f49-4bbc-8dbc-7122700832c0


https://github.com/Expensify/App/pull/33280#issuecomment-1921882304  is still reproducible

**Comment by adamgrzybowski** [comment]
hey @getusha should be fixed now

**Comment by getusha** [comment]
@adamgrzybowski could we add test steps ty!  🙇

**Comment by getusha** [comment]
> hey @getusha should be fixed now

Looks like it is still reproducible 😢 

https://github.com/Expensify/App/assets/75031127/93fb1a6a-5930-4247-95c9-8145b72219a6

**Comment by adamgrzybowski** [comment]
@getusha  LOL I was sure I just fixed that 😄 let me check again

**Comment by adamgrzybowski** [comment]
Are you sure you pulled last commits? I Think there were some changes in how the profile page looks 
image


But besides that I just found some issue after resizing. I am going to check that anyway.

**Comment by getusha** [comment]
>Are you sure you pulled last commits? I Think there were some changes in how the profile page looks

yes

Screenshot 2024-02-08 at 6 52 58 PM

**Comment by hayata-suenaga** [comment]
the code change looks good. The changes just remove the argument of `HOME` from the `goBack` function.

By the way, that is behavior when no router argument is provided to the function. I'm still having trouble wrapping my head around why simply removing the `Home` argument works.

**Comment by adamgrzybowski** [comment]
> By the way, that is behavior when no router argument is provided to the function. I'm still having trouble wrapping my head around why simply removing the Home argument works.

Before the HOME was a route of root stack. In that case going back to home would simply navigate to HOME and it would discard all screens above.

Now the HOME route is inside the BottomTabNavigator. For some reason instead of navigating to existing HOME this action will push a new BottomTabNavigator with a new HOME screen. That's why it looks so strange.

Current architecture assumes there is only one BottomTabNavigator.

**Comment by adamgrzybowski** [comment]
@getusha 

> yes

Is it possible that the version you tried didn't reload correctly? I don't see the gears icon in the top left corner. It should be visible after merging the latest main.

**Comment by getusha** [comment]
@adamgrzybowski 

https://github.com/Expensify/App/assets/75031127/2b37a8e9-16e4-42b0-99b1-e391be5bc45b

anyway i think it is low value bug and could be handled later. wdyt?

**Review by getusha — APPROVED** [review]
LGTM, the ^ issue is reproducible for me but looks like low value bug that can be handled outside.
@mountiny @hayata-suenaga

**Comment by getusha** [comment]
@adamgrzybowski looks like we've got a merge conflict

**Review by mountiny — DISMISSED** [review]
Yeah I think that looks good to me for now, all yours @hayata-suenaga

**Review by hayata-suenaga — DISMISSED** [review]
🟢

**Comment by hayata-suenaga** [comment]
oh... merge conflicts... @adamgrzybowski

**Comment by adamgrzybowski** [comment]
I think this time it was fixed for good 🤞 🤞 @getusha @mountiny

**Review by hayata-suenaga — APPROVED** [review]
🟢

**Review by mountiny — APPROVED** [review]
Thank you!

File: src/pages/iou/request/step/IOURequestStepDescription.js
Comment (Beamanator) [inline src/pages/iou/request/step/IOURequestStepDescription.js:75]: I thinkkkk getting this piece to staging will fix these blockers:
- https://github.com/Expensify/App/issues/36306
- https://github.com/Expensify/App/issues/36307

Comment (Beamanator) [reply src/pages/iou/request/step/IOURequestStepDescription.js:75]: CP'd just this to staging yesterday to unblock deploy, in https://github.com/Expensify/App/pull/36329

**Pull request merged**
